api/txspool.go

```diff
@@ -64,6 +64,72 @@ func (a *API) postPoolTx(c *gin.Context) {
 	c.JSON(http.StatusOK, receivedTx.TxID.String())
 }
 
+func (a *API) putPoolTx(c *gin.Context) {
+	txID, err := parsers.ParsePoolTxFilter(c)
+	if err != nil {
+		retBadReq(&apiError{
+			Err:  err,
+			Code: ErrParamValidationFailedCode,
+			Type: ErrParamValidationFailedType,
+		}, c)
+		return
+	}
+	var receivedTx common.PoolL2Tx
+	if err := c.ShouldBindJSON(&receivedTx); err != nil {
+		retBadReq(&apiError{
+			Err:  err,
+			Code: ErrParamValidationFailedCode,
+			Type: ErrParamValidationFailedType,
+		}, c)
+		return
+	}
+
+	receivedTx.TxID = txID
+
+	if isAtomic(receivedTx) {
+		retBadReq(&apiError{
+			Err:  errors.New(ErrNotAtomicTxsInPostPoolTx),
+			Code: ErrNotAtomicTxsInPostPoolTxCode,
+			Type: ErrNotAtomicTxsInPostPoolTxType,
+		}, c)
+		return
+	}
+	if err := a.verifyPoolL2Tx(receivedTx); err != nil {
+		retBadReq(err, c)
```

Comment:
Why this error doesn't use the apiError structure?

Reply:
bcs internally in verifyPoolL2Tx function we already have apiError structure